### ISSUE TITLE
Realtime UI updates using PubSub

### ISSUE THREAD

**Issue by amos-kibet** [3Stones-io/omedis#420]
In some resource actions, we update the records, but don't propagate changes to UI. The user has to refresh the page to see them. 
We can achieve realtime UI updates by broadcasting the changes in the resources, and subscribing to them on the live views.

**Resource-specific child issues:**

- [x] #434
- [x] #480
- [x] #481

**Comment by amos-kibet** [comment]
The PR for this issue may become too big. I propose to create child issues per resource, so that the PRs can be focused on only one resource and its live views. This way as well, I don't have to do all the work; we can split it up with @kagure-nyakio, whenever she does not have other work. For a start, I'm going to create an invitation-pubsub child issue.

Also, do we show flash messages whenever something on the UI is changed via PubSub? I imagine it may "make noise" to the users, but I also imagine a situation where a user is on the screen, reading through something, and then it "magically" changes without their notice. I don't know if that can be an issue. What can we do about this?

**Comment by almirsarajcic** [comment]
I'd broadcast only so that it's useful for showing changes in the UI. That way, we'll avoid unnecessary work.

I don't get the part about flash messages. Don't we already show flash messages when a user submits any form?

**Comment by amos-kibet** [comment]
You are right, for changes done from the UI, the flash messages are shown. Sorry I did not make it clearer. What I meant is for changes that happen on the background, like an invitation getting expired.

**Comment by amos-kibet** [comment]
But also, the flash messages are showing up on the browser where the form submission happens. I have two browsers open, logged in as the same user, and navigated to the invitations page on both browsers. When I create an invitation, it immediately shows up on both browsers (because of PubSub), but I only see the flash message on the browser that I used to create the invitation.

**Comment by almirsarajcic** [comment]
We shouldn't show flash messages for stuff from the background.
Ideally, we should send emails for such events.
For email expiry, I wouldn't do it, though.

For the second comment, a flash message should appear only in the browser submitting the form.

**Comment by almirsarajcic** [comment]
@kagure-nyakio, can you confirm this will be done when the Events PubSub PR gets merged?
Is there anything else we should cover?

**Comment by kagure-nyakio** [comment]
It will be done.
